erpnext/buying/doctype/supplier_scorecard_criteria/supplier_scorecard_criteria.py

```
@@ -73,22 +73,16 @@ def _get_variables(criteria):
-				my_variables.append(var)
-			except Exception:
-				frappe.throw(
-					_("Unable to find variable:") + " " + str(match.group(1)), InvalidFormulaVariable
+				variables = frappe.get_all(
+					"Supplier Scorecard Variable",
+					fields=["variable_label", "description", "param_name", "path"],
+					filters={"param_name": param_name},
+					limit=1,
 				)
+				my_variables.append(variables[0])
+			except (IndexError, Exception):
```

Ruff (0.15.13)

[warning] Do not catch blind exception: `Exception`

(BLE001)